src/main/java/ch/saunah/saunahbackend/service/BookingService.java

```diff
@@ -0,0 +1,143 @@
+package ch.saunah.saunahbackend.service;
+
+import ch.saunah.saunahbackend.dto.BookingBody;
+import ch.saunah.saunahbackend.model.Booking;
+import ch.saunah.saunahbackend.model.BookingState;
+import ch.saunah.saunahbackend.model.Price;
+import ch.saunah.saunahbackend.model.Sauna;
+import ch.saunah.saunahbackend.repository.BookingRepository;
+import ch.saunah.saunahbackend.repository.PriceRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.webjars.NotFoundException;
+
+import java.util.Date;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * This class contains the booking service methods
+ */
+@Service
+public class BookingService {
+
+    @Autowired
+    private BookingRepository bookingRepository;
+    @Autowired
+    private PriceRepository priceRepository;
+    @Autowired
+    private SaunaService saunaService;
+
+    /**
+     * Add a new booking to the database and assigns the end price to it.
+     *
+     * @param bookingBody the required parameters for creating a booking
+     * @return the newly created booking object
+     * @throws NullPointerException if the required object is null
+     */
+    public Booking addBooking(BookingBody bookingBody) throws NullPointerException {
+        Objects.requireNonNull(bookingBody, "BookingBody must not be null!");
```

Comment:
check if other fields which are required in the database are not null, for example bookingdate.. etc.